internal/database/repositories/statistics_repository.go

```diff
@@ -0,0 +1,173 @@
+package repositories
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/l3montree-dev/devguard/internal/core"
+	"github.com/l3montree-dev/devguard/internal/database"
+
+	"github.com/l3montree-dev/devguard/internal/database/models"
+)
+
+type statisticsRepository struct {
+	db database.DB
+}
+
+func NewStatisticsRepository(db core.DB) *statisticsRepository {
+
+	return &statisticsRepository{
+		db: db,
+	}
+}
+
+func (g statisticsRepository) GetFlawsDetailsByAssetId(assetID uuid.UUID) ([]models.Flaw, error) {
+	var flaws []models.Flaw = []models.Flaw{}
+	if err := g.db.Preload("Events").Find(&flaws, "asset_id = ?", assetID).Error; err != nil {
+		return nil, err
+	}
+	return flaws, nil
+}
+
+func (r *statisticsRepository) GetRecentFlawsForAsset(assetID uuid.UUID, time time.Time) ([]models.FlawRisk, error) {
+	var flawRisk []models.FlawRisk
+
+	if err := r.db.Raw(`
+		WITH RankedEvents AS (
+			SELECT 
+				flaw_events.flaw_id, 
+				flaw_events.created_at,
+				flaw_events.arbitrary_json_data,
+				flaw_events.type,
+				ROW_NUMBER() OVER (PARTITION BY flaw_events.flaw_id ORDER BY flaw_events.created_at DESC) AS rn
+			FROM 
+				flaw_events
+			WHERE 
+				flaw_events.created_at <= ?
+				AND EXISTS (
+					SELECT 1 
+					FROM flaws 
+					WHERE flaws.id = flaw_events.flaw_id 
+					  AND flaws.asset_id = ?
+				)
+				AND flaw_events.type IN ('detected', 'rawRiskAssessmentUpdated')
+		)
+		SELECT 
+			flaw_id, 
+			created_at, 
+			arbitrary_json_data,
+			type
+		FROM 
+			RankedEvents
+		WHERE 
+			rn = 1
+		ORDER BY 
+			flaw_id;
+	`, time, assetID).Scan(&flawRisk).Error; err != nil {
+
+		return nil, err
+	}
+
+	return flawRisk, nil
+}
+
+func (r *statisticsRepository) GetRecentFlawsState(assetID uuid.UUID, time time.Time) ([]models.FlawRisk, error) {
+	var flawRisk []models.FlawRisk
+
+	if err := r.db.Raw(`
+		WITH RankedEvents AS (
+			SELECT 
+				flaw_events.flaw_id, 
+				flaw_events.created_at,
+				flaw_events.arbitrary_json_data,
+				flaw_events.type,
+				ROW_NUMBER() OVER (PARTITION BY flaw_events.flaw_id ORDER BY flaw_events.created_at DESC) AS rn
+			FROM 
+				flaw_events
+			WHERE 
+				flaw_events.created_at <= ?
+				AND EXISTS (
+					SELECT 1 
+					FROM flaws 
+					WHERE flaws.id = flaw_events.flaw_id 
+					  AND flaws.asset_id = ?
+				)
+				AND flaw_events.type IN ('detected', 'reopened', 'fixed', 'accepted', 'falsePositive')
+		)
+		SELECT 
+			flaw_id, 
+			created_at, 
+			arbitrary_json_data,
+			type
+		FROM 
+			RankedEvents
+		WHERE 
+			rn = 1
+		ORDER BY 
+			flaw_id;
+	`, time, assetID).Scan(&flawRisk).Error; err != nil {
+
+		return nil, err
+	}
+
+	return flawRisk, nil
+}
+
+func (r *statisticsRepository) GetAssetCriticalDependenciesGroupedByScanType(asset_ID string) ([]models.AssetCriticalDependencies, error) {
```

Comment:
I think the name is misleading. You are returning all just a scanner_id and a count. Basically the amount of dependencies per scan type. You are not returning the critical dependencies of the asset - thats what the name implies